ingestion/ingestion/ingest.py

```diff
@@ -181,24 +181,35 @@ def validate_and_upload_data(
 def prune_bucket(bucket: storage.Bucket | FakeBucket):
     """Prune the bucket and waits until the bucket is empty by checking it periodically."""
 
-    bucket.lifecycle_rules = [{"action": {"type": "Delete"}, "condition": {"age": 0}}]
-    bucket.patch()
+    yesterday = datetime.now(timezone.utc) - timedelta(days=1)
+    prune_rule = {
+        "action": {"type": "Delete"},
+        "condition": {"createdBefore": yesterday.strftime("%Y-%m-%d")},
+    }
+
+    def is_prune_lifecycle(rule: dict) -> bool:
+        return (
+            rule["action"]["type"] == "Delete"
+            and "createdBefore" in rule["condition"]
+            and len(rule["condition"].keys()) == 0
```

Comment:
It's not necessary to check the size on the keys, you can have the `len` of a dict in python, so `len(rule["condition"]) == 0` should do it :)

Reply:
yeh, I removed it the last commit, don't know why it still showed up in diff xD